fbw-a32nx/src/systems/fmgc/src/guidance/vnav/VerticalProfileComputationParameters.ts

```diff
@@ -145,6 +147,26 @@ export class VerticalProfileComputationParametersObserver {
     );
   }
 
+  private getV2Speed(): Knots {
+    const fmV2 = this.fmgc.getV2Speed();
+    if (Number.isFinite(fmV2)) {
+      return fmV2;
+    }
+
+    const fmZfw = UnitType.POUND.convertTo(this.fmgc.getZeroFuelWeight(), UnitType.TONNE);
+    const fmFob = this.fmgc.getFOB();
+    const fmGw = fmZfw + fmFob;
```

Comment:
Use getGrossWeight(). It handles all the edge case and gives you the right value, or null if it isn't possible (then you will need to use some nominal value?).